src/mount/direntry_cache.h

```diff
@@ -487,14 +513,27 @@ class DirEntryCache {
 		std::unique_lock<SharedMutex> guard(rwlock_);
 		// lookup_set_ should contain all the elements inside index_set
 		auto it = lookup_set_.lower_bound(
-		    std::make_tuple(parent_inode, 0, 0, ""), LookupCompare());
+		    std::make_tuple(parent_inode, ctx.uid, ctx.gid, ""), LookupCompare());
 		while (it != lookup_set_.end() &&
 		       std::make_tuple(parent_inode, ctx.uid, ctx.gid) ==
 		               std::make_tuple(it->parent_inode, it->uid, it->gid)) {
 			DirEntry *entry = std::addressof(*it);
 			++it;
 			erase(entry);
 		}
+
+		// Make sure inode_multiset_ is also clean of outdated entries.
+		// There are scenarios where we can not determine yet the inodes from
+		// the parent, like in unlink
+		auto iter = inode_multiset_.lower_bound(parent_inode, InodeCompare());
+
+		while (iter != inode_multiset_.end() &&
+		       (iter->inode == parent_inode ||
+		        iter->parent_inode == kInvalidParent)) {
+			DirEntry *entry = std::addressof(*iter);
+			++iter;
+			erase(entry);
+		}
```

Comment:
See my previous comment. Consider adding mocked data inside ```inode_multiset``` container to test this new implementation.

Reply:
It will be addressed in this task:
https://3.basecamp.com/4375752/buckets/30999151/card_tables/cards/7271353671